pinot-controller/src/main/java/org/apache/pinot/controller/api/access/AccessType.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.controller.api.access;
+
+/**
+ * Different access types used in access control of the rest endpoints
+ */
+public enum AccessType {
+  CREATE, READ, UPDATE, DELETE
```

Comment:
I wonder if there's a way of combining CREATE and UPDATE into one. I can see a case for update without create, but create without update seems tricky (e.g. create vs update with instance partitions)

Reply:
I see your point. It's a bit tricky. I went with 4 basic CRUD operations and to annotate endpoints, I tried to make a call to see if the endpoint actually creates something or it just update the state of the system.